lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java

```diff
@@ -139,25 +159,81 @@ public static String queryTaskSummariesDb2() {
         + CLOSING_SCRIPT_TAG;
   }
 
+  @SuppressWarnings("unused")
+  public static String queryTaskSummariesOracle() {
+    return OPENING_SCRIPT_TAG
+        + "SELECT "
+        + commonSelectFields()
+        + "<if test=\"addAttachmentColumnsToSelectClauseForOrdering\">"
+        + ", a.CLASSIFICATION_ID, a.CLASSIFICATION_KEY, a.CHANNEL, a.REF_VALUE, a.RECEIVED"
+        + "</if>"
+        + "<if test=\"addClassificationNameToSelectClauseForOrdering\">, c.NAME </if>"
+        + "<if test=\"addAttachmentClassificationNameToSelectClauseForOrdering\">, ac.NAME </if>"
+        + "<if test=\"addWorkbasketNameToSelectClauseForOrdering\">, w.NAME </if>"
+        + "<if test=\"joinWithUserInfo\">, u.LONG_NAME </if>"
+        + "FROM TASK t "
+        + "<if test=\"joinWithAttachments\">LEFT JOIN ATTACHMENT a ON t.ID = a.TASK_ID </if>"
+        + "<if test=\"joinWithSecondaryObjectReferences\">LEFT JOIN OBJECT_REFERENCE o "
+        + "ON t.ID = o.TASK_ID </if>"
+        + "<if test=\"joinWithClassifications\">LEFT JOIN CLASSIFICATION c "
+        + "ON t.CLASSIFICATION_ID = c.ID </if>"
+        + "<if test=\"joinWithAttachmentClassifications\">LEFT JOIN CLASSIFICATION ac "
+        + "ON a.CLASSIFICATION_ID = ac.ID </if>"
+        + "<if test=\"joinWithWorkbaskets\">LEFT JOIN WORKBASKET w ON t.WORKBASKET_ID = w.ID </if>"
+        + "<if test=\"joinWithUserInfo\">LEFT JOIN USER_INFO u ON t.owner = u.USER_ID </if>"
+        + "WHERE t.rowid = (SELECT <if test=\"useDistinctKeyword\">DISTINCT</if> t2.rowid "
+        + "FROM TASK t2 "
+        + "<if test=\"joinWithAttachments\">LEFT JOIN ATTACHMENT a2 ON t2.ID = a2.TASK_ID </if>"
+        + "<if test=\"joinWithSecondaryObjectReferences\">LEFT JOIN OBJECT_REFERENCE o2 "
+        + "ON t2.ID = o2.TASK_ID </if>"
+        + "<if test=\"joinWithClassifications\">LEFT JOIN CLASSIFICATION c2 "
+        + "ON t2.CLASSIFICATION_ID = c2.ID </if>"
+        + "<if test=\"joinWithAttachmentClassifications\">LEFT JOIN CLASSIFICATION ac2 "
+        + "ON a2.CLASSIFICATION_ID = ac2.ID </if>"
+        + "<if test=\"joinWithWorkbaskets\">LEFT JOIN WORKBASKET w2 "
+        + "ON t2.WORKBASKET_ID = w2.ID </if>"
+        + "<if test=\"joinWithUserInfo\">LEFT JOIN USER_INFO u2 ON t2.owner = u2.USER_ID </if>"
```

Comment:
I know that we have to do something specific for this oracle db since the database locking only works for a specific case. Can we document that? 
Is there a way to reduce duplications? This way future adjustments have to be done in this method aswell, which is a little bit unfortunate

Reply:
i wrote the following comment:

```
you cant lock a view in oracle. 
the sql code `FETCH FIRST ROW ONLY` would create in oracle a view therefore we must first select a rowid based on where criteria then we select everything based on rowid and lock this rowid
```

Reply:
remove duplications is not possible for me because I don't see duplications